:libs:elasticsearch-x-content:generateProviderImpl failed: path too long

I don't seem to get a build scan out of this, but after running various builds a few times I got this error:

```
:libs:elasticsearch-x-content:generateProviderImpl (Thread[Execution worker,5,main]) started.
Detected overflow for /Users/davidturner/src/elasticsearch-main/libs/x-content/build/generated-resources/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content
Overflow detected (type: OPERATING_SYSTEM) for watched path '/Users/davidturner/src/elasticsearch-main/libs/x-content/build/generated-resources/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content', invalidating

> Task :libs:elasticsearch-x-content:generateProviderImpl FAILED
Caching disabled for task ':libs:elasticsearch-x-content:generateProviderImpl' because:
  Build cache is disabled
  Not worth caching
Task ':libs:elasticsearch-x-content:generateProviderImpl' is not up-to-date because:
  Task has failed previously.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':libs:elasticsearch-x-content:generateProviderImpl'.
> Could not copy file '/Users/davidturner/src/elasticsearch-main/libs/x-content/build/generated-resources/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/jackson-dataformat-smile-2.15.0.jar/META-INF' to '/Users/davidturner/src/elasticsearch-main/libs/x-content/build/generated-resources/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/jackson-dataformat-smile-2.15.0.jar/META-INF'.
   > Failed to create directory '/Users/davidturner/src/elasticsearch-main/libs/x-content/build/generated-resources/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/impl/IMPL-JARS/x-content/jackson-dataformat-smile-2.15.0.jar/META-INF'

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --debug option to get more log output.
> Get more help at https://help.gradle.org.

BUILD FAILED in 4s
26 actionable tasks: 1 executed, 25 up-to-date
```

A `./gradlew clean` fixed it for me. I'm on a small PR branch based on a recent main (9794c6e205e6070f1930725f7833e3e8fa18fc6a), with no build-system changes in the PR.

--- a/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/GenerateProviderManifest.java
+++ b/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/GenerateProviderManifest.java
@@ -11,14 +11,9 @@ package org.elasticsearch.gradle.internal;
 import org.elasticsearch.gradle.util.FileUtils;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.file.ConfigurableFileCollection;
-import org.gradle.api.file.DirectoryProperty;
-import org.gradle.api.file.RegularFile;
 import org.gradle.api.file.RegularFileProperty;
-import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.Classpath;
-import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.InputFiles;
-import org.gradle.api.tasks.OutputDirectory;
 import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
 

--- a/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/GenerateProviderManifest.java
+++ b/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/GenerateProviderManifest.java
@@ -12,11 +12,14 @@ import org.elasticsearch.gradle.util.FileUtils;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.file.ConfigurableFileCollection;
 import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.file.RegularFileProperty;
 import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.Classpath;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.InputFiles;
 import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
 
 import java.io.File;
@@ -33,15 +36,13 @@ abstract class GenerateProviderManifest extends DefaultTask {
     @InputFiles
     abstract public ConfigurableFileCollection getProviderImplClasspath();
 
-    @Input
-    abstract public Property<String> getManifestName();
-
-    @OutputDirectory
-    abstract DirectoryProperty getOutputDir();
+    @OutputFile
+    abstract RegularFileProperty getManifestFile();
 
     @TaskAction
     void generateManifest() {
-        File manifestFile = getOutputDir().file(getManifestName().get()).get().getAsFile();
+        File manifestFile = getManifestFile().get().getAsFile();
+        manifestFile.getParentFile().mkdirs();
         FileUtils.write(manifestFile, generateManifestContent(), "UTF-8");
     }
 

--- a/libs/x-content/build.gradle
+++ b/libs/x-content/build.gradle
@@ -67,15 +67,14 @@ tasks.named("dependencyLicenses").configure {
   mapping from: /jackson-.*/, to: 'jackson'
 }
 
-File generatedResourcesDir = new File(buildDir, 'generated-resources')
+Directory generatedResourcesDir = layout.buildDirectory.dir('generated-resources').get()
 def generateProviderManifest = tasks.register("generateProviderManifest", GenerateProviderManifest.class) {
-  outputDir = layout.buildDirectory.dir('generated-resources')
-  manifestName = "LISTING.TXT"
+  manifestFile = generatedResourcesDir.file("LISTING.TXT")
   getProviderImplClasspath().from(configurations.providerImpl)
 }
 
 def generateProviderImpl = tasks.register("generateProviderImpl", Sync) {
-  destinationDir = new File(generatedResourcesDir, "impl")
+  destinationDir = generatedResourcesDir.dir("impl").getAsFile()
   into("IMPL-JARS/x-content") {
     from(configurations.providerImpl)
     from(generateProviderManifest)

